krt: initial debug interface

howardjohn opened the pull request:
This adds an initial `krtz` debug interface.

Example output:

```yaml
name: ServiceEntryWorkloads
state:
  inputCollection: ServiceEntries
  inputs:
    default/external-svc-https:
      dependencies:
      - AuthzDerivedPolicies
      - AuthzDerivedPolicies
      - MeshConfig
      - PeerAuthentications
      - PeerAuthentications
      - static
      - static
      outputs:
      - Kubernetes/networking.istio.io/ServiceEntry/default/external-svc-https/api.dropboxapi.com
      - Kubernetes/networking.istio.io/ServiceEntry/default/external-svc-https/www.googleapis.com
  outputs:
    Kubernetes/networking.istio.io/ServiceEntry/default/external-svc-https/api.dropboxapi.com:
      canonicalName: external-svc-https
      canonicalRevision: latest
      clusterId: Kubernetes
      hostname: api.dropboxapi.com
      name: external-svc-https
      namespace: default
      services:
        default/api.dropboxapi.com:
          ports:
          - servicePort: 443
            targetPort: 443
      uid: Kubernetes/networking.istio.io/ServiceEntry/default/external-svc-https/api.dropboxapi.com
      workloadName: external-svc-https
      workloadType: POD
    Kubernetes/networking.istio.io/ServiceEntry/default/external-svc-https/www.googleapis.com:
      canonicalName: external-svc-https
      canonicalRevision: latest
      clusterId: Kubernetes
      hostname: www.googleapis.com
      name: external-svc-https
      namespace: default
      services:
        default/www.googleapis.com:
          ports:
          - servicePort: 443
            targetPort: 443
      uid: Kubernetes/networking.istio.io/ServiceEntry/default/external-svc-https/www.googleapis.com
      workloadName: external-svc-https
      workloadType: POD
```

howardjohn commented on pkg/kube/krt/informer.go:
```diff
@@ -184,6 +185,7 @@ func WrapClient[I controllers.ComparableObject](c kclient.Informer[I], opts ...C
 		close(h.synced)
 		h.log.Infof("%v synced", h.name())
 	}()
+	RegisterCollectionForDebugging(h)
```
Should probably move this to collection creation and not have a global var

istio-testing commented:
@howardjohn: The following test **failed**, say `/retest` to rerun all failed tests or `/retest-required` to rerun all mandatory failed tests:

Test name | Commit | Details | Required | Rerun command
--- | --- | --- | --- | ---
lint_istio | efc729e1c194a0c864472a1eeb8b629b0dd13c04 | [link](https://prow.istio.io/view/gs/istio-prow/pr-logs/pull/istio_istio/53597/lint_istio/1848497348397764608) | true | `/test lint`

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes-sigs/prow](https://github.com/kubernetes-sigs/prow/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>
<!-- test report -->

stevenctl commented:
Can we have a debug output that shows only the collection dependencies and outputs?

hzxuzhonghu commented:
Without looking at the code it is hard to understand the meaning. Can you add a key before`default/external-svc-https:` 
\
And also for dependencies, 

```
```

howardjohn commented:
> Can you add a key before`default/external-svc-https:`

I am not sure I understand. that string is the key.. what key would we add?

hzxuzhonghu commented:
I mean like kubectl output, you can add a explicit  `kind` ServiceEntry, etc

howardjohn commented:
ah good point. let me see if I can pass the input type through. ill clean up the filters to be more clear as well

ramaraochavali commented:
What does inputs and outputs signify?

howardjohn commented:
> What does inputs and outputs signify?

https://github.com/istio/istio/blob/915ac2bf05dbc7a08ae61a8bc39fcd6ea1f3d11a/pilot/pkg/serviceregistry/kube/controller/ambient/workloads.go#L134

Input: WorkloadEntrys
Output: WorkloadInfos

howardjohn commented:
OK I have implemented all the TODOs and cleaned things up a bit, this is ready to go from my perspective. See the original description for the new output

istio-testing commented:
@howardjohn: The following tests **failed**, say `/retest` to rerun all failed tests or `/retest-required` to rerun all mandatory failed tests:

Test name | Commit | Details | Required | Rerun command
--- | --- | --- | --- | ---
lint_istio | efc729e1c194a0c864472a1eeb8b629b0dd13c04 | [link](https://prow.istio.io/view/gs/istio-prow/pr-logs/pull/istio_istio/53597/lint_istio/1848497348397764608) | true | `/test lint`
gencheck_istio | 156e8eb98f5799565f4107a4877bc462c85fa0b9 | [link](https://prow.istio.io/view/gs/istio-prow/pr-logs/pull/istio_istio/53597/gencheck_istio/1851652291963129856) | true | `/test gencheck`

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes-sigs/prow](https://github.com/kubernetes-sigs/prow/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>
<!-- test report -->

istio-testing commented:
@howardjohn: The following tests **failed**, say `/retest` to rerun all failed tests or `/retest-required` to rerun all mandatory failed tests:

Test name | Commit | Details | Required | Rerun command
--- | --- | --- | --- | ---
gencheck_istio | 156e8eb98f5799565f4107a4877bc462c85fa0b9 | [link](https://prow.istio.io/view/gs/istio-prow/pr-logs/pull/istio_istio/53597/gencheck_istio/1851652291963129856) | true | `/test gencheck`
lint_istio | 156e8eb98f5799565f4107a4877bc462c85fa0b9 | [link](https://prow.istio.io/view/gs/istio-prow/pr-logs/pull/istio_istio/53597/lint_istio/1851652320639586304) | true | `/test lint`
integ-distroless_istio | 156e8eb98f5799565f4107a4877bc462c85fa0b9 | [link](https://prow.istio.io/view/gs/istio-prow/pr-logs/pull/istio_istio/53597/integ-distroless_istio/1851652299659677696) | true | `/test integ-distroless`

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes-sigs/prow](https://github.com/kubernetes-sigs/prow/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>
<!-- test report -->

istio-testing commented:
@howardjohn: The following tests **failed**, say `/retest` to rerun all failed tests or `/retest-required` to rerun all mandatory failed tests:

Test name | Commit | Details | Required | Rerun command
--- | --- | --- | --- | ---
gencheck_istio | 156e8eb98f5799565f4107a4877bc462c85fa0b9 | [link](https://prow.istio.io/view/gs/istio-prow/pr-logs/pull/istio_istio/53597/gencheck_istio/1851652291963129856) | true | `/test gencheck`
lint_istio | 156e8eb98f5799565f4107a4877bc462c85fa0b9 | [link](https://prow.istio.io/view/gs/istio-prow/pr-logs/pull/istio_istio/53597/lint_istio/1851652320639586304) | true | `/test lint`
integ-distroless_istio | 156e8eb98f5799565f4107a4877bc462c85fa0b9 | [link](https://prow.istio.io/view/gs/istio-prow/pr-logs/pull/istio_istio/53597/integ-distroless_istio/1851652299659677696) | true | `/test integ-distroless`
integ-telemetry_istio | 156e8eb98f5799565f4107a4877bc462c85fa0b9 | [link](https://prow.istio.io/view/gs/istio-prow/pr-logs/pull/istio_istio/53597/integ-telemetry_istio/1851652319796531200) | true | `/test integ-telemetry`

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes-sigs/prow](https://github.com/kubernetes-sigs/prow/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>
<!-- test report -->

istio-testing commented:
@howardjohn: The following tests **failed**, say `/retest` to rerun all failed tests or `/retest-required` to rerun all mandatory failed tests:

Test name | Commit | Details | Required | Rerun command
--- | --- | --- | --- | ---
gencheck_istio | 156e8eb98f5799565f4107a4877bc462c85fa0b9 | [link](https://prow.istio.io/view/gs/istio-prow/pr-logs/pull/istio_istio/53597/gencheck_istio/1851652291963129856) | true | `/test gencheck`
lint_istio | 156e8eb98f5799565f4107a4877bc462c85fa0b9 | [link](https://prow.istio.io/view/gs/istio-prow/pr-logs/pull/istio_istio/53597/lint_istio/1851652320639586304) | true | `/test lint`
integ-distroless_istio | 156e8eb98f5799565f4107a4877bc462c85fa0b9 | [link](https://prow.istio.io/view/gs/istio-prow/pr-logs/pull/istio_istio/53597/integ-distroless_istio/1851652299659677696) | true | `/test integ-distroless`
integ-telemetry_istio | 156e8eb98f5799565f4107a4877bc462c85fa0b9 | [link](https://prow.istio.io/view/gs/istio-prow/pr-logs/pull/istio_istio/53597/integ-telemetry_istio/1851652319796531200) | true | `/test integ-telemetry`
unit-tests_istio | 156e8eb98f5799565f4107a4877bc462c85fa0b9 | [link](https://prow.istio.io/view/gs/istio-prow/pr-logs/pull/istio_istio/53597/unit-tests_istio/1851652327354667008) | true | `/test unit-tests`

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes-sigs/prow](https://github.com/kubernetes-sigs/prow/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>
<!-- test report -->

istio-testing commented:
@howardjohn: The following tests **failed**, say `/retest` to rerun all failed tests or `/retest-required` to rerun all mandatory failed tests:

Test name | Commit | Details | Required | Rerun command
--- | --- | --- | --- | ---
gencheck_istio | 156e8eb98f5799565f4107a4877bc462c85fa0b9 | [link](https://prow.istio.io/view/gs/istio-prow/pr-logs/pull/istio_istio/53597/gencheck_istio/1851652291963129856) | true | `/test gencheck`
lint_istio | 156e8eb98f5799565f4107a4877bc462c85fa0b9 | [link](https://prow.istio.io/view/gs/istio-prow/pr-logs/pull/istio_istio/53597/lint_istio/1851652320639586304) | true | `/test lint`
integ-distroless_istio | 156e8eb98f5799565f4107a4877bc462c85fa0b9 | [link](https://prow.istio.io/view/gs/istio-prow/pr-logs/pull/istio_istio/53597/integ-distroless_istio/1851652299659677696) | true | `/test integ-distroless`
integ-telemetry_istio | 156e8eb98f5799565f4107a4877bc462c85fa0b9 | [link](https://prow.istio.io/view/gs/istio-prow/pr-logs/pull/istio_istio/53597/integ-telemetry_istio/1851652319796531200) | true | `/test integ-telemetry`
unit-tests_istio | 156e8eb98f5799565f4107a4877bc462c85fa0b9 | [link](https://prow.istio.io/view/gs/istio-prow/pr-logs/pull/istio_istio/53597/unit-tests_istio/1851652327354667008) | true | `/test unit-tests`
macbuildcheck_istio | 156e8eb98f5799565f4107a4877bc462c85fa0b9 | [link](https://prow.istio.io/view/gs/istio-prow/pr-logs/pull/istio_istio/53597/macbuildcheck_istio/1851652330701721600) | true | `/test macbuildcheck`
integ-cni_istio | 156e8eb98f5799565f4107a4877bc462c85fa0b9 | [link](https://prow.istio.io/view/gs/istio-prow/pr-logs/pull/istio_istio/53597/integ-cni_istio/1851652293896704000) | true | `/test integ-cni`
integ-ipv6_istio | 156e8eb98f5799565f4107a4877bc462c85fa0b9 | [link](https://prow.istio.io/view/gs/istio-prow/pr-logs/pull/istio_istio/53597/integ-ipv6_istio/1851652304697036800) | true | `/test integ-ipv6`

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes-sigs/prow](https://github.com/kubernetes-sigs/prow/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>
<!-- test report -->

istio-testing commented:
@howardjohn: The following tests **failed**, say `/retest` to rerun all failed tests or `/retest-required` to rerun all mandatory failed tests:

Test name | Commit | Details | Required | Rerun command
--- | --- | --- | --- | ---
gencheck_istio | 156e8eb98f5799565f4107a4877bc462c85fa0b9 | [link](https://prow.istio.io/view/gs/istio-prow/pr-logs/pull/istio_istio/53597/gencheck_istio/1851652291963129856) | true | `/test gencheck`
lint_istio | 156e8eb98f5799565f4107a4877bc462c85fa0b9 | [link](https://prow.istio.io/view/gs/istio-prow/pr-logs/pull/istio_istio/53597/lint_istio/1851652320639586304) | true | `/test lint`
integ-distroless_istio | 156e8eb98f5799565f4107a4877bc462c85fa0b9 | [link](https://prow.istio.io/view/gs/istio-prow/pr-logs/pull/istio_istio/53597/integ-distroless_istio/1851652299659677696) | true | `/test integ-distroless`
integ-telemetry_istio | 156e8eb98f5799565f4107a4877bc462c85fa0b9 | [link](https://prow.istio.io/view/gs/istio-prow/pr-logs/pull/istio_istio/53597/integ-telemetry_istio/1851652319796531200) | true | `/test integ-telemetry`
unit-tests_istio | 156e8eb98f5799565f4107a4877bc462c85fa0b9 | [link](https://prow.istio.io/view/gs/istio-prow/pr-logs/pull/istio_istio/53597/unit-tests_istio/1851652327354667008) | true | `/test unit-tests`
macbuildcheck_istio | 156e8eb98f5799565f4107a4877bc462c85fa0b9 | [link](https://prow.istio.io/view/gs/istio-prow/pr-logs/pull/istio_istio/53597/macbuildcheck_istio/1851652330701721600) | true | `/test macbuildcheck`
integ-cni_istio | 156e8eb98f5799565f4107a4877bc462c85fa0b9 | [link](https://prow.istio.io/view/gs/istio-prow/pr-logs/pull/istio_istio/53597/integ-cni_istio/1851652293896704000) | true | `/test integ-cni`
integ-ipv6_istio | 156e8eb98f5799565f4107a4877bc462c85fa0b9 | [link](https://prow.istio.io/view/gs/istio-prow/pr-logs/pull/istio_istio/53597/integ-ipv6_istio/1851652304697036800) | true | `/test integ-ipv6`
integ-telemetry-discovery_istio | 156e8eb98f5799565f4107a4877bc462c85fa0b9 | [link](https://prow.istio.io/view/gs/istio-prow/pr-logs/pull/istio_istio/53597/integ-telemetry-discovery_istio/1851652313081450496) | true | `/test integ-telemetry-discovery`
integ-security_istio | 156e8eb98f5799565f4107a4877bc462c85fa0b9 | [link](https://prow.istio.io/view/gs/istio-prow/pr-logs/pull/istio_istio/53597/integ-security_istio/1851652310569062400) | true | `/test integ-security`
integ-ds_istio | 156e8eb98f5799565f4107a4877bc462c85fa0b9 | [link](https://prow.istio.io/view/gs/istio-prow/pr-logs/pull/istio_istio/53597/integ-ds_istio/1851652300506927104) | true | `/test integ-ds`

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes-sigs/prow](https://github.com/kubernetes-sigs/prow/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>
<!-- test report -->

istio-testing commented:
@howardjohn: The following tests **failed**, say `/retest` to rerun all failed tests or `/retest-required` to rerun all mandatory failed tests:

Test name | Commit | Details | Required | Rerun command
--- | --- | --- | --- | ---
gencheck_istio | 156e8eb98f5799565f4107a4877bc462c85fa0b9 | [link](https://prow.istio.io/view/gs/istio-prow/pr-logs/pull/istio_istio/53597/gencheck_istio/1851652291963129856) | true | `/test gencheck`
lint_istio | 156e8eb98f5799565f4107a4877bc462c85fa0b9 | [link](https://prow.istio.io/view/gs/istio-prow/pr-logs/pull/istio_istio/53597/lint_istio/1851652320639586304) | true | `/test lint`
integ-distroless_istio | 156e8eb98f5799565f4107a4877bc462c85fa0b9 | [link](https://prow.istio.io/view/gs/istio-prow/pr-logs/pull/istio_istio/53597/integ-distroless_istio/1851652299659677696) | true | `/test integ-distroless`
integ-telemetry_istio | 156e8eb98f5799565f4107a4877bc462c85fa0b9 | [link](https://prow.istio.io/view/gs/istio-prow/pr-logs/pull/istio_istio/53597/integ-telemetry_istio/1851652319796531200) | true | `/test integ-telemetry`
unit-tests_istio | 156e8eb98f5799565f4107a4877bc462c85fa0b9 | [link](https://prow.istio.io/view/gs/istio-prow/pr-logs/pull/istio_istio/53597/unit-tests_istio/1851652327354667008) | true | `/test unit-tests`
macbuildcheck_istio | 156e8eb98f5799565f4107a4877bc462c85fa0b9 | [link](https://prow.istio.io/view/gs/istio-prow/pr-logs/pull/istio_istio/53597/macbuildcheck_istio/1851652330701721600) | true | `/test macbuildcheck`
integ-cni_istio | 156e8eb98f5799565f4107a4877bc462c85fa0b9 | [link](https://prow.istio.io/view/gs/istio-prow/pr-logs/pull/istio_istio/53597/integ-cni_istio/1851652293896704000) | true | `/test integ-cni`
integ-ipv6_istio | 156e8eb98f5799565f4107a4877bc462c85fa0b9 | [link](https://prow.istio.io/view/gs/istio-prow/pr-logs/pull/istio_istio/53597/integ-ipv6_istio/1851652304697036800) | true | `/test integ-ipv6`
integ-telemetry-discovery_istio | 156e8eb98f5799565f4107a4877bc462c85fa0b9 | [link](https://prow.istio.io/view/gs/istio-prow/pr-logs/pull/istio_istio/53597/integ-telemetry-discovery_istio/1851652313081450496) | true | `/test integ-telemetry-discovery`
integ-security_istio | 156e8eb98f5799565f4107a4877bc462c85fa0b9 | [link](https://prow.istio.io/view/gs/istio-prow/pr-logs/pull/istio_istio/53597/integ-security_istio/1851652310569062400) | true | `/test integ-security`
integ-ds_istio | 156e8eb98f5799565f4107a4877bc462c85fa0b9 | [link](https://prow.istio.io/view/gs/istio-prow/pr-logs/pull/istio_istio/53597/integ-ds_istio/1851652300506927104) | true | `/test integ-ds`
integ-basic-arm64_istio | 156e8eb98f5799565f4107a4877bc462c85fa0b9 | [link](https://prow.istio.io/view/gs/istio-prow/pr-logs/pull/istio_istio/53597/integ-basic-arm64_istio/1851652293049454592) | true | `/test integ-basic-arm64`

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes-sigs/prow](https://github.com/kubernetes-sigs/prow/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>
<!-- test report -->

istio-testing commented:
@howardjohn: The following tests **failed**, say `/retest` to rerun all failed tests or `/retest-required` to rerun all mandatory failed tests:

Test name | Commit | Details | Required | Rerun command
--- | --- | --- | --- | ---
gencheck_istio | 156e8eb98f5799565f4107a4877bc462c85fa0b9 | [link](https://prow.istio.io/view/gs/istio-prow/pr-logs/pull/istio_istio/53597/gencheck_istio/1851652291963129856) | true | `/test gencheck`
lint_istio | 156e8eb98f5799565f4107a4877bc462c85fa0b9 | [link](https://prow.istio.io/view/gs/istio-prow/pr-logs/pull/istio_istio/53597/lint_istio/1851652320639586304) | true | `/test lint`
integ-distroless_istio | 156e8eb98f5799565f4107a4877bc462c85fa0b9 | [link](https://prow.istio.io/view/gs/istio-prow/pr-logs/pull/istio_istio/53597/integ-distroless_istio/1851652299659677696) | true | `/test integ-distroless`
integ-telemetry_istio | 156e8eb98f5799565f4107a4877bc462c85fa0b9 | [link](https://prow.istio.io/view/gs/istio-prow/pr-logs/pull/istio_istio/53597/integ-telemetry_istio/1851652319796531200) | true | `/test integ-telemetry`
unit-tests_istio | 156e8eb98f5799565f4107a4877bc462c85fa0b9 | [link](https://prow.istio.io/view/gs/istio-prow/pr-logs/pull/istio_istio/53597/unit-tests_istio/1851652327354667008) | true | `/test unit-tests`
macbuildcheck_istio | 156e8eb98f5799565f4107a4877bc462c85fa0b9 | [link](https://prow.istio.io/view/gs/istio-prow/pr-logs/pull/istio_istio/53597/macbuildcheck_istio/1851652330701721600) | true | `/test macbuildcheck`
integ-cni_istio | 156e8eb98f5799565f4107a4877bc462c85fa0b9 | [link](https://prow.istio.io/view/gs/istio-prow/pr-logs/pull/istio_istio/53597/integ-cni_istio/1851652293896704000) | true | `/test integ-cni`
integ-ipv6_istio | 156e8eb98f5799565f4107a4877bc462c85fa0b9 | [link](https://prow.istio.io/view/gs/istio-prow/pr-logs/pull/istio_istio/53597/integ-ipv6_istio/1851652304697036800) | true | `/test integ-ipv6`
integ-telemetry-discovery_istio | 156e8eb98f5799565f4107a4877bc462c85fa0b9 | [link](https://prow.istio.io/view/gs/istio-prow/pr-logs/pull/istio_istio/53597/integ-telemetry-discovery_istio/1851652313081450496) | true | `/test integ-telemetry-discovery`
integ-security_istio | 156e8eb98f5799565f4107a4877bc462c85fa0b9 | [link](https://prow.istio.io/view/gs/istio-prow/pr-logs/pull/istio_istio/53597/integ-security_istio/1851652310569062400) | true | `/test integ-security`
integ-ds_istio | 156e8eb98f5799565f4107a4877bc462c85fa0b9 | [link](https://prow.istio.io/view/gs/istio-prow/pr-logs/pull/istio_istio/53597/integ-ds_istio/1851652300506927104) | true | `/test integ-ds`
integ-basic-arm64_istio | 156e8eb98f5799565f4107a4877bc462c85fa0b9 | [link](https://prow.istio.io/view/gs/istio-prow/pr-logs/pull/istio_istio/53597/integ-basic-arm64_istio/1851652293049454592) | true | `/test integ-basic-arm64`
unit-tests-arm64_istio | 156e8eb98f5799565f4107a4877bc462c85fa0b9 | [link](https://prow.istio.io/view/gs/istio-prow/pr-logs/pull/istio_istio/53597/unit-tests-arm64_istio/1851652334044581888) | true | `/test unit-tests-arm64`

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes-sigs/prow](https://github.com/kubernetes-sigs/prow/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>
<!-- test report -->

istio-testing commented:
@howardjohn: The following tests **failed**, say `/retest` to rerun all failed tests or `/retest-required` to rerun all mandatory failed tests:

Test name | Commit | Details | Required | Rerun command
--- | --- | --- | --- | ---
gencheck_istio | 156e8eb98f5799565f4107a4877bc462c85fa0b9 | [link](https://prow.istio.io/view/gs/istio-prow/pr-logs/pull/istio_istio/53597/gencheck_istio/1851652291963129856) | true | `/test gencheck`
lint_istio | 156e8eb98f5799565f4107a4877bc462c85fa0b9 | [link](https://prow.istio.io/view/gs/istio-prow/pr-logs/pull/istio_istio/53597/lint_istio/1851652320639586304) | true | `/test lint`
integ-distroless_istio | 156e8eb98f5799565f4107a4877bc462c85fa0b9 | [link](https://prow.istio.io/view/gs/istio-prow/pr-logs/pull/istio_istio/53597/integ-distroless_istio/1851652299659677696) | true | `/test integ-distroless`
integ-telemetry_istio | 156e8eb98f5799565f4107a4877bc462c85fa0b9 | [link](https://prow.istio.io/view/gs/istio-prow/pr-logs/pull/istio_istio/53597/integ-telemetry_istio/1851652319796531200) | true | `/test integ-telemetry`
unit-tests_istio | 156e8eb98f5799565f4107a4877bc462c85fa0b9 | [link](https://prow.istio.io/view/gs/istio-prow/pr-logs/pull/istio_istio/53597/unit-tests_istio/1851652327354667008) | true | `/test unit-tests`
macbuildcheck_istio | 156e8eb98f5799565f4107a4877bc462c85fa0b9 | [link](https://prow.istio.io/view/gs/istio-prow/pr-logs/pull/istio_istio/53597/macbuildcheck_istio/1851652330701721600) | true | `/test macbuildcheck`
integ-cni_istio | 156e8eb98f5799565f4107a4877bc462c85fa0b9 | [link](https://prow.istio.io/view/gs/istio-prow/pr-logs/pull/istio_istio/53597/integ-cni_istio/1851652293896704000) | true | `/test integ-cni`
integ-ipv6_istio | 156e8eb98f5799565f4107a4877bc462c85fa0b9 | [link](https://prow.istio.io/view/gs/istio-prow/pr-logs/pull/istio_istio/53597/integ-ipv6_istio/1851652304697036800) | true | `/test integ-ipv6`
integ-telemetry-discovery_istio | 156e8eb98f5799565f4107a4877bc462c85fa0b9 | [link](https://prow.istio.io/view/gs/istio-prow/pr-logs/pull/istio_istio/53597/integ-telemetry-discovery_istio/1851652313081450496) | true | `/test integ-telemetry-discovery`
integ-security_istio | 156e8eb98f5799565f4107a4877bc462c85fa0b9 | [link](https://prow.istio.io/view/gs/istio-prow/pr-logs/pull/istio_istio/53597/integ-security_istio/1851652310569062400) | true | `/test integ-security`
integ-ds_istio | 156e8eb98f5799565f4107a4877bc462c85fa0b9 | [link](https://prow.istio.io/view/gs/istio-prow/pr-logs/pull/istio_istio/53597/integ-ds_istio/1851652300506927104) | true | `/test integ-ds`
integ-basic-arm64_istio | 156e8eb98f5799565f4107a4877bc462c85fa0b9 | [link](https://prow.istio.io/view/gs/istio-prow/pr-logs/pull/istio_istio/53597/integ-basic-arm64_istio/1851652293049454592) | true | `/test integ-basic-arm64`
unit-tests-arm64_istio | 156e8eb98f5799565f4107a4877bc462c85fa0b9 | [link](https://prow.istio.io/view/gs/istio-prow/pr-logs/pull/istio_istio/53597/unit-tests-arm64_istio/1851652334044581888) | true | `/test unit-tests-arm64`
integ-helm_istio | 156e8eb98f5799565f4107a4877bc462c85fa0b9 | [link](https://prow.istio.io/view/gs/istio-prow/pr-logs/pull/istio_istio/53597/integ-helm_istio/1851652296417480704) | true | `/test integ-helm`
integ-ambient_istio | 156e8eb98f5799565f4107a4877bc462c85fa0b9 | [link](https://prow.istio.io/view/gs/istio-prow/pr-logs/pull/istio_istio/53597/integ-ambient_istio/1851652295490539520) | true | `/test integ-ambient`
integ-pilot-istiodremote_istio | 156e8eb98f5799565f4107a4877bc462c85fa0b9 | [link](https://prow.istio.io/view/gs/istio-prow/pr-logs/pull/istio_istio/53597/integ-pilot-istiodremote_istio/1851652301337399296) | true | `/test integ-pilot-istiodremote`

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes-sigs/prow](https://github.com/kubernetes-sigs/prow/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>
<!-- test report -->

istio-testing commented:
@howardjohn: The following tests **failed**, say `/retest` to rerun all failed tests or `/retest-required` to rerun all mandatory failed tests:

Test name | Commit | Details | Required | Rerun command
--- | --- | --- | --- | ---
integ-security_istio | 156e8eb98f5799565f4107a4877bc462c85fa0b9 | [link](https://prow.istio.io/view/gs/istio-prow/pr-logs/pull/istio_istio/53597/integ-security_istio/1851652310569062400) | true | `/test integ-security`
integ-ambient_istio | 156e8eb98f5799565f4107a4877bc462c85fa0b9 | [link](https://prow.istio.io/view/gs/istio-prow/pr-logs/pull/istio_istio/53597/integ-ambient_istio/1851652295490539520) | true | `/test integ-ambient`
lint_istio | 3f070f51064154e30fbd96ec47b8a1afca25e3c2 | [link](https://prow.istio.io/view/gs/istio-prow/pr-logs/pull/istio_istio/53597/lint_istio/1851653241226399744) | true | `/test lint`
integ-distroless_istio | 3f070f51064154e30fbd96ec47b8a1afca25e3c2 | [link](https://prow.istio.io/view/gs/istio-prow/pr-logs/pull/istio_istio/53597/integ-distroless_istio/1851653222779850752) | true | `/test integ-distroless`

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes-sigs/prow](https://github.com/kubernetes-sigs/prow/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>
<!-- test report -->

hzxuzhonghu commented on pkg/kube/krt/collection.go:
```diff
@@ -159,24 +159,55 @@ func (h *manyCollection[I, O]) Synced() Syncer {
+	//for os, o := range h.collectionState.outputs {
+	//	h.log.Errorf("Output %v -> %v", os, o)
+	//}
+	//h.log.Errorf("<<< END DUMP")
```
nit: remove

hzxuzhonghu commented:
default/external-svc-https:    
      dependencies:
      - AuthzDerivedPolicies          // can this be set to user aware istio API 
      - AuthzDerivedPolicies   // duplicate
      - MeshConfig
      - PeerAuthentications
      - PeerAuthentications
      - static                       // what does static mean
      - static

howardjohn commented:
> ```
> default/external-svc-https:    
>   dependencies:
>   - AuthzDerivedPolicies          // can this be set to user aware istio API 
>   - AuthzDerivedPolicies   // duplicate
>   - MeshConfig
>   - PeerAuthentications
>   - PeerAuthentications
>   - static                       // what does static mean
>   - static 
> ```

oops, my bad - I fixed this already but forgot to update the description after fixing that one. Now it says NetworkTrigger I believe

stevenctl commented on pkg/kube/krt/debug.go:
```diff
@@ -0,0 +1,81 @@
+
+// maybeRegisterCollectionForDebugging registers the collection in the debugger, if one is enabled
+func maybeRegisterCollectionForDebugging[T any](c Collection[T], handler *DebugHandler) {
+	if handler == nil {
```
it could be nice to have an env like `KRT_GLOBAL_DEBUG` that allows enabling debug on all collections without modifying code to apss in `GlobalDebugHandler`
